src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        firstNode = null;
+        lastNode = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            firstNode = newNode;
+            lastNode = newNode;
+        } else {
+            newNode.prev = lastNode;
+            lastNode.next = newNode;
+            lastNode = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || size < index) {
+            throw new IndexOutOfBoundsException("Index out of range: " + index);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
```

Comment:
it would be better to create a separate method that would add node at the beginning or in the middle.